latex, make use of ``\small`` for code listings customizable

jfbu opened the pull request:
Until now, latex forces use of `\small` in code-blocks. This is explainable by the fact that
1. the default setting of courier for typewriter font gives big glyphs
2. until recently Sphinx did not know how to wrap code lines

This PR maintains same result as current Sphinx, but the user can override the default.

jfbu committed:
latex, make use of ``\small`` for code listings customizable

tk0miya commented on doc/config.rst:
```diff
@@ -1663,6 +1663,16 @@ These options influence LaTeX output. See further :doc:`latex`.
      ``'maketitle'``
         "maketitle" call, default ``'\\maketitle'``.  Override if you want to
         generate a differently-styled title page.
+     ``'fvset'``
```
The settings is like a contract to users.
The documentation has a very long life. So we should keep a compatibility of the new settings on feature sphinx.
I think we have to be careful to add them.

We already have `preamble` key to customize these trivial settings.
So I think no special key is not needed here.

jfbu replied:
@tk0miya do you mean that `\fvset{fontsize=\small}` should be part of `sphinx.sty` and advanced user can still do in `preamble` key own setting for `\fvset` ? Perhaps then move mention of `\fvset` to the new `latex.rst`. You suggest suppression of `'fvset'` key ?

tk0miya commented:
LGTM :-)

jfbu commented:
@tk0miya I removed the `'fvset'` new key and moved the default `\fvset{fontsize=\small}` (which replaces the formerly hardcoded `\small` in Verbatim) to inside `sphinx.sty`. Documentation is moved to `latex.rst` for explaining to user the new customizability possibility. There is no functional change in how Sphinx operates by default.

tk0miya commented:
Thank you for changing. LGTM.

jfbu commented:
rebased and squashed before merge

jfbu commented:
@tk0miya thanks for reviewing !
